tests/infrastructure/test_docker_container.py

```diff
@@ -689,3 +691,36 @@ def test_run_requires_command():
     container = DockerContainer(command=[])
     with pytest.raises(ValueError, match="cannot be run with empty command"):
         container.run()
+
+def test_stream_container_logs(capsys, mock_docker_client):
+    mock_container = mock_docker_client.containers.get.return_value
+    
+    mock_container.logs = MagicMock(return_value=[b"hello", b"world"])
+    
+    DockerContainer(
+        command=["doesnt", "matter"],
+    ).run()
+    
+    captured = capsys.readouterr()
+    
+    assert captured.out == 'hello\nworld\n'
+    
+def test_logs_warning_when_container_marked_for_removal(caplog, mock_docker_client):
+    mock_container = mock_docker_client.containers.get.return_value
+    mock_container.logs = MagicMock(side_effect=docker.errors.APIError("...marked for removal..."))
```

Comment:
It might be good to use the full error message we see from Docker here instead of the specific substring we know we're looking for.